chef/cookbooks/crowbar/templates/default/crowbar-pre-upgrade.sh.erb

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+#
+# This does what is necessary directly before the start of OS upgrade
+
+LOGFILE=/var/log/crowbar/node-upgrade.log
+mkdir -p "`dirname "$LOGFILE"`"
+
+# Copy stdout to fd 3
+exec 3>&1
+# Create fd 4 for logfile
+exec 4>> "$LOGFILE"
+# Set fd 1 and 2 to logfile
+exec 1>&4 2>&1
```

Comment:
Yeah - it's wrong to copy and paste that stuff from `install-chef-suse.sh` to here, because that has a totally different approach to output (a user-friendly ASCII-graphics display using `dialog(1)`, spinners etc., plus more verbose logging to files), whereas here there is no pseudo-graphical UI.